.github/workflows/build.yml

```diff
@@ -60,23 +60,35 @@ jobs:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          extensions: ${{ matrix.operating-system == 'windows-latest' && contains('["8.2","8.3"]', matrix.php) && 'sockets, curl, zip, ffi' || 'sockets, curl, zip, ffi, grpc' }}
+          extensions: 'sockets, curl, zip, ffi'
           php-version: ${{ matrix.php }}
           coverage: none
-          ini-values: ${{ matrix.operating-system == 'windows-latest' && 'opcache.enable=0 opcache.enable_cli=0' || '' }}
+          ini-values: ${{ matrix.os == 'windows-2022' && 'opcache.enable=0 opcache.enable_cli=0' || '' }}
 
       - name: Install Protoc
         uses: arduino/setup-protoc@v2
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Install gRPC Extension (for PHP 8.2 & 8.3 on Windows)
+      - name: Install gRPC Extension
         run: |
-          cd C:\tools\php
-          Invoke-WebRequest -Uri https://github.com/tienvx/build-php-grpc-extension/releases/download/builds/grpc_windows-2022_php-${{ matrix.php }}.dll -OutFile ext\php_grpc.dll
-          echo "extension=php_grpc.dll" >> php.ini
-        if: ${{ matrix.operating-system == 'windows-latest' && contains('["8.2","8.3"]', matrix.php) }}
-        shell: pwsh
+          if ! [ -x "$(command -v php-config)" ]
+          then
+            ext_dir='C:\tools\php\ext'
+            ini_path='C:\tools\php\php.ini'
+            sudo=
+            lib_name=php_grpc.dll
+            ext=dll
+          else
+            ext_dir=$(php-config --extension-dir)
+            ini_path=$(php-config --ini-dir)/grpc.ini
+            sudo=sudo
+            lib_name=grpc.so
+            ext=so
+          fi
+          $sudo curl -L -o $ext_dir/$lib_name https://github.com/tienvx/build-php-grpc-extension/releases/download/builds/grpc_${{ matrix.os }}_php-${{ matrix.php }}.$ext
+          echo "extension=$lib_name" | $sudo tee -a $ini_path
```

Comment:
I have concerns about this. Also why all the renaming of operating-system to os?

I think a pre-requisite is that the grpc extension come into the PACT Foundation and is audited by someone that understands what that binary blob does.

Reply:
> Also why all the renaming of operating-system to os?

Because it's shorter when it's repeated multiple times in the PR.

> I think a pre-requisite is that the grpc extension come into the PACT Foundation and is audited by someone that understands what that binary blob does.

The code is open source at https://github.com/tienvx/build-php-grpc-extension.
I think it's not worth continue doing this. I will close this PR.
I will also remove this github action step when grpc extension binary for PHP 8.2 & 8.3 on windows is available again.